integration/volume/volume_test.go

```diff
@@ -0,0 +1,115 @@
+package volume
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	volumetypes "github.com/docker/docker/api/types/volume"
+	"github.com/docker/docker/integration/internal/container"
+	"github.com/docker/docker/integration/internal/request"
+	"github.com/docker/docker/internal/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestVolumesList(t *testing.T) {
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	prefix, _ := getPrefixAndSlashFromDaemonPlatform()
+
+	id := container.Run(t, ctx, client, container.WithVolume(prefix+"foo"))
+
+	c, err := client.ContainerInspect(ctx, id)
+	require.NoError(t, err)
+	vname := c.Mounts[0].Name
+
+	names := func(volumes []*types.Volume) []string {
+		entries := []string{}
+		for _, vol := range volumes {
+			entries = append(entries, vol.Name)
+		}
+		return entries
+	}
+	volumes, err := client.VolumeList(ctx, filters.Args{})
+	require.NoError(t, err)
+	assert.Contains(t, names(volumes.Volumes), vname)
```

Comment:
Checking the list contains a specific name is a pretty weak assertion.

Can we use a label and filter so that we can use `assert.Equal()` ?
Can we compare the entire struct instead of just names?

Reply:
Thanks @dnephin. I updated the PR so that now all elements (except for create time) is compared against.